plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -1150,3 +1150,91 @@ FROM (
 	ORDER BY [record_id] DESC
 ) AS z
 `
+
+const sqlServerAvailabilityReplicaStates string = `
+IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) BEGIN /*NOT IN Standard,Enterpris,Express*/
```

Comment:
Can you also add this check - aka collect only if availability groups enabled

IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4)  or SERVERPROPERTY('IsHadrEnabled') = 0

Reply:
Fixed !

Reply:
reverted this change and did the IsHadrEnabled check before executing the query